brightStars_DM_sprint_Feb2020/BeyondRadius.ipynb

```diff
@@ -0,0 +1,422 @@
+    "The radius of current HSC bright star masks, made by Andy Goulding, are selected by comparing the density of detections in increasingly large annuli around each bright stars with the average detection density. \n",
+    "\n",
+    "Still, one of the identified problems with the HSC Public Data Release 2 is that false detections sometimes occur around the edge of these masks. The aim of this notebook is thus to provide a plot of the number of detections as a function of increasing radius, around bright stars and beyond the edge of their mask."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Options\n",
+    "One might want to change the two cells below.\n",
+    "\n",
+    "The first one allows one to choose the path to the Butler repo we will read in, filter (note it only changes masks read in*, since we read detections from merged catalog), tracts, and binning options for the annuli.\n",
+    "\n",
+    "The rerun in `/project/mschmitz/hscRerun/mynobg` is a subset of the 25 central patches in tract 9697, and had local background subtraction turned off.\n",
+    "\n",
+    "Normally, we'd select bright objects to study from an outside catalog (Gaia). But for now, we are just reading in the actual mask region files, which means we have to jump through a few extra hoops.\n",
+    "If we want to look at whole tracts, we can just set `reselectBrightObjects` to `False`; if further selection of bright objects is needed, pass on a string.\n",
+    "The only one recognized at the moment is `boxAround`, where we pick a rectangular box around the furthest detections (because I only reran on the inner 25 patches of a tract for analysis sprint). This is admittedly hacky, but it's temporary, while we work with Andy's masks (and might prove useful anyway, eg to do this plot on the whole COSMOS field, that is circular).\n",
+    "\n",
+    "\\* not even that in the case of the Arcturus masks, that are all those from the I band despite the filenames containing a filter."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "butlerPath = \"/project/mschmitz/hscRerun/mynobg\"\n",
+    "\n",
+    "chosenFilter = \"HSC-R\" \n",
+    "tractList = [9697] \n",
+    "patchList = [[\"{},{}\".format(i,j) for i in range(2,7) for j in range(2,7)]]\n",
+    "\n",
+    "# Annuli parameters\n",
+    "annSize = 5 # radius of each consecutive annulus, in arcsec\n",
+    "nbAnnuli = 20 # number of annuli\n",
+    "\n",
+    "# which detections to keep\n",
+    "reselectBrightObjects = \"boxAround\"#False\n",
+    "\n",
+    "# print a few checkpoints, validation plots, etc\n",
+    "verbose = False\n",
+    "\n",
+    "# which mask to load - if False, loads Arcturus masks\n",
+    "gouldingMask = True"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Change this cell if you want to save the end plot:"
```

Comment:
"Adapt the following based on whether you want to save the end plot:"?

Allow for the user to select an `outputDir` here?   Default it to `plots/` and use the variable in cell [8].